shardus_net/src/shardus_net_listener.rs

```diff
@@ -34,31 +34,31 @@ pub enum ListenerError {
 type ListenerResult<T> = Result<T, ListenerError>;
 
 impl ShardusNetListener {
-    pub fn new<A: ToSocketAddrs>(address: A) -> Result<Self, ()> {
+    pub fn new<A: ToSocketAddrs>(address: A, net_config: NetConfig) -> Result<Self, ()> {
         let mut addresses = address.to_socket_addrs().map_err(|_| ())?;
         let address = addresses.next().ok_or(())?;
 
-        Ok(Self { address })
+        Ok(Self { address, net_config })
     }
 
     pub fn listen(&self) -> UnboundedReceiver<(String, SocketAddr, Option<RequestMetadata>)> {
-        Self::spawn_listener(self.address)
+        Self::spawn_listener(self.address, self.net_config.clone())
```

Comment:
we can avoid cloning here, we only need to make a copy when we need to pass ownership to different threads or services.

Reply:
Checking. The goal is to make it pass-by-value everywhere but I was getting compilation error with that.

Reply:
Made the changes, please check.